app/scripts/models/auth_brokers/fx-sync-channel.js

```diff
@@ -268,14 +268,22 @@ define(function (require, exports, module) {
       return ! _.difference(REQUIRED_LOGIN_FIELDS, loginFields).length;
     },
 
+    /**
+     * Get login data from `account` to send to the browser.
+     * All returned keys have a defined value.
+     *
+     * @param {Object} account
+     * @returns {Object}
+     * @private
+     */
     _getLoginData (account) {
       const loginData = account.pick(ALLOWED_LOGIN_FIELDS);
       loginData.verified = !! loginData.verified;
       loginData.verifiedCanLinkAccount = !! this._verifiedCanLinkEmail &&
                                          !! this._verifiedCanLinkUid;
-      return loginData;
-    }
 
+      return _.omit(loginData, _.isUndefined);
```

Comment:
I noticed this was a bug previously. `_hasRequiredLoginFields` counted fields that had `undefined` as a value, because of this bug, we had this workaround for afterPasswordResetConfirmationPoll in fx-desktop-v2](https://github.com/mozilla/fxa-content-server/pull/5472/files#diff-a1ed3eb721932d97da56f4a5117ed165L49). By fixing this bug, the workaround can be removed.